### ISSUE TITLE
Visualize dependency tree

### ISSUE THREAD

**Issue by savente93** [prefix-dev/pixi#457]
### Problem description

As an upstream library/runtime for some applications I would like to keep my dependency tree as lean as possible. Currently this is quite cumbersome with conda. Conda tree helps for visualization why certain dependencies are included (e.g. `conda-tree whoneeds mydep` and I would love to have this just as is, but what would be really great is to also visualize the size of the dependencies with it. Currently conda has no support for this as far as I can tell meaning I have to hack it together with scripts. I don't mind if this is a slow operation but it would be a very nice feature for upstream applications.

**Comment by ruben-arts** [comment]
Good idea, I think this would fit with the #238 issue. As there are some overlapping mechanisms. 

Also all info is already in the lock file. So I would be easy to get it out of there (for all conda dependencies). I asked chatgpt for a small example:
```python
import yaml
import pandas as pd


# Parse the YAML data
with open("/home/rarts/development/pixi/pixi.lock", 'r') as file:
    yaml_data = yaml.safe_load(file)

# Extract the relevant information
packages = yaml_data['package']
names_sizes = []
for pkg in packages:
    if pkg["manager"] != "pip" and pkg["name"] and pkg["size"]:
        names_sizes.append((pkg["name"], pkg["size"]))

# Create a DataFrame
df = pd.DataFrame(names_sizes, columns=['Name', 'Size'])

# Calculate total size
total_size = df['Size'].sum()

# Display the DataFrame and total size
print(df)
print(f"Total Size: {total_size}")
```
